[IMP] sale_promotion_gift: bugs reportados por github. Cantidades por plantilla, producto y categoria

agaldona opened the pull request.

dulhaver committed:
changed wording to "Secondary Unit of Purchase"

original wording "Unit of Purchase" results into fields 
 - Purchase Unit of Measure    (from   product.template)   &
 - Unit of Purchase   &
appearing in the "procurement" tab in the products from view

I think it differentiates a little clearer between those 2 with a new wording "Secondary Unit of Purchase"

dulhaver committed:
reduced length of line to below 80

line 29, line 34

Commit:
[ADD] mrp_repair_stock_picking_out

Commit:
[ADD] mrp_repair_stock_picking_out

Commit:
[IMP] Modifications according to comments

Commit:
[IMP] Modifications according to comments

Commit:
Merge branch 'mrp_repair_stock_picking_out' of https://github.com/Daniel-CA/odoomrp-wip-public into mrp_repair_stock_picking_out

Commit:
[IMP] Modifications in prepare method

Commit:
[IMP] Short license in files

Commit:
[IMP] stock_quant_manual_assign: Delete picking operations to confirm the wizard of "quants Manual" and check availability.

sysadminmatmoz committed:
Added string to view field to let it be translated

sysadminmatmoz committed:
Fixed wrong file

Commit:
[IMP] mrp_production_capacity: If quantity produced is greater than the capacity per cycle, the necessary production orders are created to cover the amount of produce

am-technix committed:
French translation of stock_inventory_preparation_filters

pedrobaeza committed:
Merge pull request #893 from am-technix/8.0

French translation of stock_inventory_preparation_filters

pedrobaeza committed:
[IMP] README: Add Runbot flag

pedrobaeza committed:
Merge pull request #890 from sysadminmatmoz/8.0

Added string to view field to let it being translated

Commit:
[FIX] stock_quant_manual_assign: Improve programming.

Commit:
[IMP] purchase_mrp_project_link: borrar las vistas que hacen invisible la cuenta analitica en las líneas de pedido de compra

pedrobaeza committed:
Merge pull request #894 from agaldona/8.0-purcahse_analytic_line_view

[IMP] purchase_mrp_project_link: borrar las vistas que hacen invisible la cuenta analitica en las líneas de pedido de compra

Commit:
[ADD] mrp_production_simulated_capacity: nuevo módulo para obtener la cantidad minima a la hora de crear la OF ficticia.

Commit:
[IMP] mrp_production_simulated_capacity: correcciones solicitadas

Commit:
[IMP] mrp_production_project_estimated_cost: onchange, antes y despues de obtener la cantidad.

oihane committed:
[ADD] oca_dependencies.txt added

Commit:
[IMP] mrp_repair_stock_picking_out: Warehouse added for stock.picking.type search

Commit:
Merge pull request #857 from Daniel-CA/mrp_repair_stock_picking_out

[ADD] mrp_repair_stock_picking_out

pedrobaeza committed:
Merge pull request #895 from agaldona/8.0-simulated_production_capacitu

[ADD] mrp_production_simulated_capacity: nuevo módulo para obtener la cantidad minima a la hora de crear la OF ficticia

Commit:
[FIX] stock_quant_manual_assign: Improve programming.

pedrobaeza committed:
Merge pull request #880 from alfredoavanzosc/odoomrp-stock_quant_manual_assign

[IMP] stock_quant_manual_assign: Delete picking operations to confirm…

Commit:
[FIX] mrp_production_capacity: Improved programming.

Commit:
[FIX] procurement_plan_mrp: Display error if not defined the product in BoM line.

pedrobaeza committed:
Merge pull request #897 from alfredoavanzosc/odoomrp-procurement_plan_mrp

[FIX] procurement_plan_mrp: Display error if not defined the product …

Commit:
[IMP] sale_promotion_gift: bugs reportados por github. Cantidades por plantilla, producto y categoria

Commit:
[IMP] mrp_production_project_estimated_cost: producto en el tree de analitica.

pedrobaeza committed:
Merge pull request #898 from agaldona/8.0-product_in_analytic_lines

[IMP] mrp_production_project_estimated_cost: producto en el tree de analitica

agaldona commented:
@pedrobaeza no se si te llegan las notificaciones por defecto en este caso, de todas formas, entiendo que cuando puedas tendrías que echar un vistazo a este PR, y cuando este ok para merge, lo tendría que mergear @alfredoavanzosc si no me equivoco.

pedrobaeza commented on sale_promotion_gift/models/sale_order.py:
```diff
@@ -13,8 +13,10 @@ class SaleOrder(models.Model):
         lines_with_price = self.order_line.filtered(lambda x: x.price_unit > 0)
-        self.sale_promotion_gifts = [(
-            6, 0, lines_with_price.offer_id.sale_promotion_gifts.mapped('id'))]
+        gifts_list = []
```
Esto debe funcionar sin problemas, por qué lo cambias?

agaldona replied:
Por que cuando hay mas de una línea de pedido de venta con precio, me da un expected singleton.
Al parecer, cuando lines_with_price es una lista no lo hace bien. Deduje que era porque hacia .offer_id y en ese caso solo espera que haya uno

pedrobaeza replied:
Sí, prueba esto: `lines_with_price.mapped('offer_id.sale_promotion_gifts.mapped.id')`

pedrobaeza commented on sale_promotion_gift/models/sale_order.py:
```diff
@@ -48,14 +51,51 @@ def button_dump_sale_final_gifts(self):
         for line in self.sale_promotion_gifts:
-            vals = self._prepare_final_gift_product_data(self.id, line)
-            self.env['sale.final.gift'].create(vals)
+            vals = self._prepare_final_gift_product_data(self, line)
```
Por qué se pasa otra vez self?

pedrobaeza commented on sale_promotion_gift/models/product_pricelist_item_offer.py:
```diff
@@ -11,3 +11,4 @@ class ProductPricelistItemOffer(models.Model):
     sale_promotion_gifts = fields.One2many(
         'sale.promotion.gift', 'product_pricelist_item_offer',
         string='Sale promotion gifts', copy=False)
+    not_combinable = fields.Boolean('Offer not combinable')
```
Esto no está documentado qué hace exactamente, y es un nuevo requisito, no un bug

pedrobaeza replied:
Vuelvo a repetir, ¿qué hace esto? Debe documentarse y es una nueva opción de fallo. ¿Por qué se incluye aquí?

agaldona replied:
He añadido la funcionalidad de ese campo en el readme. ¿También le tengo que añadir un help?
Es un campo que indica si una oferta es acumulable o no. Si en un pedido se carga una oferta por categoría, y otra por producto y ese producto pertenece esa categoría, si este campo no esta marcado, aparte de la oferta de producto, se le aplicara la oferta de categoría a la línea. En cambio, si la oferta de producto es no acumulable, no se le aplicara la de la categoría.

Commit:
[IMP] sale_promotion_gift: nuevo concepto, oferta no acumulable + correcciones.

agaldona commented:
Modificaciones realizadas.

pedrobaeza committed:
Merge pull request #892 from alfredoavanzosc/odoomrp-mrp_production_capacity

[IMP] mrp_production_capacity: If quantity produced is greater than t…

Commit:
[ADD] mrp_production_manual_close

pedrobaeza committed:
[IMP] purchase_landed_cost: Improve search of distribution and expense lines

pedrobaeza committed:
[IMP] purchase_landed_cost: Include default amount for expense types

Commit:
[IMP] account_treasury_forecast: mensaje en el botón get data Closes #652

Commit:
[IMP] mrp_production_close: Move comments to known issues section.

esthermm committed:
[ADD] New module <purchase_requisition_full_bid_order_generator>

pedrobaeza committed:
[REM/IMP] CI files

esthermm committed:
[IMP] Changes requested by Travis

esthermm committed:
[IMP] Travis

Commit:
Merge pull request #899 from Daniel-CA/mrp_production_manual_close

[ADD] mrp_production_manual_close

esthermm committed:
[IMP] Readme updated

esthermm committed:
[IMP] Optimized function

oihane committed:
[IMP] Travis errors fixed

oihane committed:
[IMP] Remove sudo

pedrobaeza committed:
Merge pull request #903 from oihane/8.0-travis-test

[IMP] Travis errors fixed

oihane committed:
Merge pull request #901 from esthermm/8.0

[ADD] New module <purchase_requisition_full_bid_order_generator>

pedrobaeza committed:
[FIX] purchase_product_variants: Lookup of existing product when confirming fixed

pedrobaeza committed:
Merge pull request #889 from alfredoavanzosc/odoomrp-sale_promotion_gift

[ADD] sale_promotion_gift: New module to put free product in the offers.

ejans committed:
changed default references of fields.Datetime.now() to fields.Datetime.now

agaldona commented:
Este PR estaría bien asi? o tengo que hacer mas modificaciones?

oihane committed:
[IMP] <mrp_bom_by_percentage> Fixed mrp.bom form view

Commit:
Update README.rst

Commit:
Updated to follow OCA conventions

Explained in
https://github.com/OCA/maintainer-tools/tree/master/template/module

Commit:
Update README.rst

Commit:
Update README.rst

ddiazpinto committed:
[FIX] purchase_product_variants: Checks the number of attributes selected in `purchase.order.line` and the corresponding `product.product` to ensure to be equal and avoid `expected singleton` error.

oihane committed:
Merge pull request #923 from oihane/8.0-percent

[IMP] <mrp_bom_by_percentage> Fixed mrp.bom form view

oihane committed:
[IMP] <mrp_production_capacity> Fix error when opening MO.

Commit:
[FIX] mrp_operations_extension Fixes #920 Now initiate a WO doesn't force MO material reservation

oihane committed:
[IMP] <mrp_production_capacity> Continue fixing things

alfredoavanzosc committed:
[ADD] sale_order_other_expenses: Define in sales orders, other expenses that are not transportation.

alfredoavanzosc committed:
[FIX] sale_orer_other_expenses: Module renamed as delivery_multi_expense.

pedrobaeza committed:
Merge pull request #926 from oihane/8.0-fix-capacity

[IMP] <mrp_production_capacity> Fix error when opening MO.

Commit:
[ADD] mrp_operations_start_without_material closes #921
mrp_operations_extension: Added function to avoid repeat code

pedrobaeza committed:
Merge pull request #924 from ddiazpinto/8.0

[FIX] purchase_product_variants: Avoids `expected singleton` error

pedrobaeza committed:
Merge pull request #900 from agaldona/8.0-account_treasury_forecast_get_data

[IMP] account_treasury_forecast: mensaje en el botón get data

pedrobaeza committed:
[FIX] account_treasury_*: Regularize translations

pedrobaeza committed:
Merge pull request #931 from Daniel-CA/mrp_operations_start_without_material

[ADD] mrp_operations_start_without_material closes #921

pedrobaeza committed:
Merge pull request #927 from Daniel-CA/mrp_operations_extension

[FIX] mrp_operations_extension Fixes #920 Now initiate a WO doesn't force MO material reservation

pedrobaeza commented:
No hecha prueba funcional, pero :+1:

pedrobaeza commented:
Se debe integrar en la rama del PR o hacer directamente un PR contra la rama principal

pedrobaeza committed:
Merge pull request #912 from ejans/8.0

changed default references of fields.Datetime.now()

pedrobaeza committed:
Merge pull request #705 from vrms/improve-product_secondary_unit

changed wording to "Secondary Unit of Purchase"

pedrobaeza committed:
[IMP] sale_product_variants: Fine grain readonly in sale order line fields

sysadminmatmoz committed:
Slovene translations added

sysadminmatmoz committed:
Slovene translations added

sysadminmatmoz committed:
Merge branch '8.0' of github.com:sysadminmatmoz/odoomrp-wip into 8.0

pedrobaeza committed:
Merge pull request #932 from sysadminmatmoz/8.0

Slovene translations added

sysadminmatmoz committed:
Translations and templates sync and fixes

sysadminmatmoz committed:
Merge branch '8.0' of github.com:sysadminmatmoz/odoomrp-wip into 8.0

pedrobaeza committed:
Merge pull request #933 from sysadminmatmoz/8.0

Translations and templates sync and fixes

alfredoavanzosc committed:
[FIX] delivery_multi_expense: Improved programming.

alfredoavanzosc committed:
[FIX] delivery_multi_expense: Improve programming.

pedrobaeza committed:
Merge pull request #928 from alfredoavanzosc/odoomrp-sale_order_other_expenses

[ADD] delivery_multi_expense: Define in sales orders, other expens…

pedrobaeza committed:
[IMP] delivery_multi_expense: Improve README

Commit:
[FIX] mrp_configurable timing: llamada correcta al super en la funcion _get_workorder_in_product_lines
[FIX] mrp_product_variants_configurable_timing: llamada super en funcion _get_workorder_in_product_lines

pedrobaeza committed:
Merge pull request #935 from agaldona/8.0-timing_configurable

[FIX] mrp_configurable timing: llamada correcta al super en la función _get_workorder_in_product_lines

Commit:
[FIX] sale_promotion_gift: calculo de importe a regalar, multiplicar por unidades regaladas por cada pack
